app/scripts/metamask-controller.js

```diff
@@ -756,7 +756,8 @@ module.exports = class MetamaskController extends EventEmitter {
     try {
       const threeBoxSyncingAllowed = this.threeBoxController.getThreeBoxSyncingState()
       if (threeBoxSyncingAllowed && !this.threeBoxController.box) {
-        await this.threeBoxController.new3Box()
+        // 'await' intentionally omitted to avoid waiting for initialization
+        this.threeBoxController.init()
```

Comment:
I think we need to wait for `this.threeBoxController.init()` before calling `this.threeBoxController.turnThreeBoxSyncingOn()` or else `_update3Box()` could be called before `this.space` is defined

Reply:
Right, good point.

Reply:
Wait, shouldn't `threeBoxSynced` take care of that? `update3Box` won't do anything until after `threeBoxSynced` is true, which would imply that the initialization was finished

Reply:
Yeah looking at the flow, `threeBoxSynced` will block using `this.space` until it's set